### PR TITLE
MGMT-11763-enable TPMv2

### PR DESCRIPTION
**Pull request by alknopfler**
Signed-off-by: Alberto Morgante Medina <alknopfler@users.noreply.github.com>

# Description

This PR enables the TPMv2 feature for hub and edgecluster in order to have the disks encripted

- Hub requirement (virtualized environment):
   - Install ```swtpm``` package in the hypervisor (`yum install -y swtpm`)
   - Libvirt configuration: 
          device section:
```
          <tpm model='tpm-tis'>
                 <backend type='emulator' version='2.0'/>
          </tpm>
```
_The doc section has been updated to explain this in requirements section_

Basically, with the requirement the PR will install tpm and use it for the hub and edgeclusters   

Fixes # [MGMT-11763](https://issues.redhat.com/browse/MGMT-11763)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

**HUB tests:**

image
image

**Edgecluster tests:**
image
image
image
image

**Test Configuration**:

- Versions: 4.10.20 and 4.10.30
- Hardware: virtualize (flavio3)

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: deploy-edgecluster/render_edgeclusters.sh [hunk 1]
```diff
@@ -128,8 +128,11 @@ metadata:
   name: $CHANGE_EDGE_NAME
   namespace: $CHANGE_EDGE_NAME
   annotations:
-      agent-install.openshift.io/install-config-overrides: '{"networking":{"networkType": "OpenshiftSDN"}}'
+      agent-install.openshift.io/install-config-overrides: '{"networking":{"networkType":"OpenshiftSDN"}}'
 spec:
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply deploy-edgecluster/render_edgeclusters.sh:132]: after talking yesterday with Flavio, it's something we want

**Review by danielchg — APPROVED** [review]
LGTM

File: deploy-edgecluster/render_edgeclusters.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (flaper87) [reply deploy-edgecluster/render_edgeclusters.sh:132]: Correct! Let's keep this enable by default.

Having said this, we will all have to re-deploy our hub clusters

[the remark this answers is not in the document]
Comment (flaper87) [reply deploy-edgecluster/render_edgeclusters.sh:132]: Disregard the last comment, this is just for the edge, we should be ok.

**Pull request merged**